builtin/logical/database/path_rotate_credentials.go

```diff
@@ -144,6 +151,21 @@ func (b *databaseBackend) pathRotateRoleCredentialsUpdate() framework.OperationF
 			return nil, err
 		}
 
+		// Send a canary to force a guard check which prevents the client
+		// to run in a timeout if the request was originally send to a
+		// performance secondary/standby node.
+		if b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary | consts.ReplicationPerformanceStandby) {
```

Comment:
Is this necessary since we are bailing out early above?

Reply:
Or should this be run on a performance primary?

Reply:
My initial idea was to only write to the storage if we are part of a performance cluster (to avoid unnecessary writes if we are not part of a cluster) but I guess that is not really important.